test/e2e/httpproxy/httpproxy_test.go

```diff
@@ -282,6 +282,116 @@ var _ = Describe("HTTPProxy", func() {
 		})
 	})
 
+	f.NamespacedTest("httpproxy-backend-tls-version", func(namespace string) {
+		BeforeEach(func() {
+			// Top level issuer.
+			selfSignedIssuer := &certmanagerv1.Issuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "selfsigned",
+				},
+				Spec: certmanagerv1.IssuerSpec{
+					IssuerConfig: certmanagerv1.IssuerConfig{
+						SelfSigned: &certmanagerv1.SelfSignedIssuer{},
+					},
+				},
+			}
+			require.NoError(f.T(), f.Client.Create(context.TODO(), selfSignedIssuer))
+
+			// CA to sign backend certs with.
+			caCertificate := &certmanagerv1.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "ca-cert",
+				},
+				Spec: certmanagerv1.CertificateSpec{
+					IsCA: true,
+					Usages: []certmanagerv1.KeyUsage{
+						certmanagerv1.UsageSigning,
+						certmanagerv1.UsageCertSign,
+					},
+					CommonName: "ca-cert",
+					SecretName: "ca-cert",
+					IssuerRef: certmanagermetav1.ObjectReference{
+						Name: "selfsigned",
+					},
+				},
+			}
+			require.NoError(f.T(), f.Client.Create(context.TODO(), caCertificate))
+
+			// Issuer based on CA to generate new certs with.
+			basedOnCAIssuer := &certmanagerv1.Issuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "ca-issuer",
+				},
+				Spec: certmanagerv1.IssuerSpec{
+					IssuerConfig: certmanagerv1.IssuerConfig{
+						CA: &certmanagerv1.CAIssuer{
+							SecretName: "ca-cert",
+						},
+					},
+				},
+			}
+			require.NoError(f.T(), f.Client.Create(context.TODO(), basedOnCAIssuer))
+
+			// Backend client cert, can use for upstream validation as well.
+			backendClientCert := &certmanagerv1.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "backend-client-cert",
+				},
+				Spec: certmanagerv1.CertificateSpec{
+					Usages: []certmanagerv1.KeyUsage{
+						certmanagerv1.UsageClientAuth,
+					},
+					CommonName: "client",
+					SecretName: "backend-client-cert",
+					IssuerRef: certmanagermetav1.ObjectReference{
+						Name: "ca-issuer",
+					},
+				},
+			}
+			require.NoError(f.T(), f.Client.Create(context.TODO(), backendClientCert))
+
+			contourConfig.TLS = config.TLSParameters{
+				ClientCertificate: config.NamespacedName{
+					Namespace: namespace,
+					Name:      "backend-client-cert",
+				},
+			}
+
+			contourConfiguration.Spec.Envoy.ClientCertificate = &contour_api_v1alpha1.NamespacedName{
+				Name:      "backend-client-cert",
+				Namespace: namespace,
+			}
+		})
+		Context("with backend tls version configured via Contour ConfigMap", func() {
```

Comment:
I think we can combine these two contexts (configmap and contourconfig-specific wrappers) and just set the same options in the BeforeEach just as we do here: https://github.com/projectcontour/contour/blob/4d48db7a095888cdec109bc4f90f87a65013082b/test/e2e/httpproxy/httpproxy_test.go#L115-L118

we have different test runs for using each type of config so we should only need one instance of the test

Reply:
okay I combined them into 1 before each. Does it look how you're imagining now?